src/m365/teams/commands/meeting/meeting-create.ts

```diff
@@ -0,0 +1,206 @@
+import { aadUser } from '../../../../utils/aadUser.js';
+import { accessToken } from '../../../../utils/accessToken.js';
+import { validation } from '../../../../utils/validation.js';
+import GraphCommand from "../../../base/GraphCommand.js";
+import commands from '../../commands.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { OnlineMeeting } from '@microsoft/microsoft-graph-types';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  startTime?: string;
+  endTime?: string;
+  subject?: string;
+  participants?: string;
+  organizerEmail?: string;
+  recordAutomatically?: boolean;
+}
+
+class TeamsMeetingCreateCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MEETING_CREATE;
+  }
+
+  public get description(): string {
+    return 'Create a new online meeting';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['subject', 'startDateTime', 'endDateTime', 'joinUrl', 'recordAutomatically'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        startTime: typeof args.options.startTime !== 'undefined',
+        endTime: typeof args.options.endTime !== 'undefined',
+        subject: typeof args.options.subject !== 'undefined',
+        participants: typeof args.options.participants !== 'undefined',
+        organizerEmail: typeof args.options.organizerEmail !== 'undefined',
+        recordAutomatically: !!args.options.recordAutomatically
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-s --startTime [startTime]'
+      },
+      {
+        option: '-e --endTime [endTime]'
+      },
+      {
+        option: '-s --subject [subject]'
+      },
+      {
+        option: '-p --participants [participants]'
+      },
+      {
+        option: '--organizerEmail [organizerEmail]'
+      },
+      {
+        option: '-r --recordAutomatically'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.startTime && !validation.isValidISODateTime(args.options.startTime)) {
+          return `'${args.options.startTime}' is not a valid ISO date string for startTime.`;
+        }
+        if (args.options.endTime && !validation.isValidISODateTime(args.options.endTime)) {
+          return `'${args.options.endTime}' is not a valid ISO date string for endTime.`;
+        }
+        if (args.options.startTime && args.options.endTime && new Date(args.options.startTime) >= new Date(args.options.endTime)) {
+          return 'startTime value must be before endTime.';
+        }
+        if (args.options.endTime && !args.options.startTime) {
```

Comment:
I think we can remove this statement, the startTime would default to the current time,

Reply:
Ok, I removed it and in a later stage added the check to set the start date to the current date when the end date is specified. Otherwise, the Graph API returns an error.